From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Sun, 16 Jul 2023 19:25:00 +0200
Subject: [PATCH] Client perspective - Block state visualization - BlockState
 field for visualization setupper

Adds a field in `BlockState` to hold the visualization setupper for that state.

Does not implement anything that sets or gets this value.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

diff --git a/src/main/java/net/minecraft/world/level/block/state/BlockState.java b/src/main/java/net/minecraft/world/level/block/state/BlockState.java
index 664572ad64ed53386a7348b34ec2bac25e9381d5..6a61097667a730379b7613989f247a155a4b5b1d 100644
--- a/src/main/java/net/minecraft/world/level/block/state/BlockState.java
+++ b/src/main/java/net/minecraft/world/level/block/state/BlockState.java
@@ -50,4 +50,13 @@ public class BlockState extends BlockBehaviour.BlockStateBase {
     }
     // Fiddle end - cache registry indices as fields - block state
 
+    // Fiddle start - client perspective - block state visualization - BlockState field for visualization setupper
+    /**
+     * The {@linkplain org.fiddlemc.fiddle.packet.block.BlockStateVisualizationSetupper}
+     * that can set up the visualizations for this {@link BlockState}.
+     * If null, there exist no replacements for this state, and it is always sent to clients as-is.
+     */
+    public @org.jetbrains.annotations.Nullable org.fiddlemc.fiddle.packet.block.BlockStateVisualizationSetupper visualizationSetupper = null;
+    // Fiddle end - client perspective - block state visualization - BlockState field for visualization setupper
+
 }
